pywr/_core.pyx

```diff
@@ -1080,4 +1085,38 @@ cdef class VirtualStorage(Storage):
             self._flow[i] = 0.0
             for n, f in zip(self._nodes, self._factors):
                 self._flow[i] -= f*n._flow[i]
-        Storage.after(self, ts)
+        Storage.after(self, ts, adjustment = adjustment)
+
+
+cdef class RollingVirtualStorage(VirtualStorage):
+    def __cinit__(self):
+        self._memory_pointer = 0
+
+    cpdef setup(self, model):
+        super(RollingVirtualStorage, self).setup(model)
+        cdef int ncomb = len(model.scenarios.combinations)
+        if self.timesteps < 2:
+            raise ValueError('The number of time-steps for a RollingVirtualStorage node must be greater than one.')
+        self._memory = np.zeros((self.timesteps-1, ncomb), dtype=np.float64)
+        self._memory_pointer = 0
+
+    cpdef reset(self):
+        VirtualStorage.reset(self)
+        self._memory[:] = 0.0
+        self._memory_pointer = 0
+
+    cpdef after(self, Timestep ts, double[:] adjustment = None):
+        cdef int i
+        cdef ScenarioIndex si
+
+        assert adjustment is None
+
+        # Update the storage volumes by applying an adjustment
+        VirtualStorage.after(self, ts, adjustment=self._memory[self._memory_pointer, :])
+
+        # Store today's flow in the memory and increment the memory pointer
+        for i, si in enumerate(self.model.scenarios.combinations):
+            # Flow is negative in VirtualStorage to remove from the store, save the +ve number here for
+            # returning to the store later
+            self._memory[self._memory_pointer, i] = -self._flow[i]
```

Comment:
Think this needs to be: 

`self._memory[self._memory_pointer, i] = -self._flow[i] * ts.day`

Otherwise the node does not work correctly if the model timestepper  delta > 1.  Probably worth having a test to check this